javacord-api/src/main/java/org/javacord/api/entity/intent/Intent.java

```diff
@@ -0,0 +1,219 @@
+     *     <li>GUILD_BAN_ADD</li>
+     *     <li>GUILD_BAN_REMOVE</li>
+     * </ul>
+     */
+    GUILD_BANS(2),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>GUILD_EMOJIS_UPDATE</li>
+     * </ul>
+     */
+    GUILD_EMOJIS(3),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>GUILD_INTEGRATIONS_UPDATE</li>
+     * </ul>
+     */
+    GUILD_INTEGRATIONS(4),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>WEBHOOKS_UPDATE</li>
+     * </ul>
+     */
+    GUILD_WEBHOOKS(5),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>INVITE_CREATE</li>
+     *     <li>INVITE_DELETE</li>
+     * </ul>
+     */
+    GUILD_INVITES(6),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>VOICE_STATE_UPDATE</li>
+     * </ul>
+     */
+    GUILD_VOICE_STATES(7),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>PRESENCE_UPDATE</li>
+     * </ul>
+     *
+     * <p>Note: This is a privileged intent which must be enabled in the bots application page in the Developer portal
+     *
+     * @see <a href="https://discordapp.com/developers/applications">Discord developer portal</a>
+     */
+    GUILD_PRESENCES(8),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>MESSAGE_CREATE</li>
+     *     <li>MESSAGE_UPDATE</li>
+     *     <li>MESSAGE_DELETE</li>
+     *     <li>MESSAGE_DELETE_BULK</li>
+     * </ul>
+     */
+    GUILD_MESSAGES(9),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>MESSAGE_REACTION_ADD</li>
+     *     <li>MESSAGE_REACTION_REMOVE</li>
+     *     <li>MESSAGE_REACTION_REMOVE_ALL</li>
+     *     <li>MESSAGE_REACTION_REMOVE_EMOJI</li>
+     * </ul>
+     */
+    GUILD_MESSAGE_REACTIONS(10),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>TYPING_START</li>
+     * </ul>
+     */
+    GUILD_MESSAGE_TYPING(11),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>CHANNEL_CREATE</li>
+     *     <li>MESSAGE_CREATE</li>
+     *     <li>MESSAGE_UPDATE</li>
+     *     <li>MESSAGE_DELETE</li>
+     *     <li>CHANNEL_PINS_UPDATE</li>
+     * </ul>
+     */
+    DIRECT_MESSAGES(12),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>MESSAGE_REACTION_ADD</li>
+     *     <li>MESSAGE_REACTION_REMOVE</li>
+     *     <li>MESSAGE_REACTION_REMOVE_ALL</li>
+     *     <li>MESSAGE_REACTION_REMOVE_EMOJI</li>
+     * </ul>
+     */
+    DIRECT_MESSAGE_REACTIONS(13),
+
+    /**
+     * The following events are received.
+     * <ul>
+     *     <li>TYPING_START</li>
+     * </ul>
+     */
+    DIRECT_MESSAGE_TYPING(14);
+
+    private final int id;
+
+    /**
+     * Class constructor.
+     *
+     * @param id The id of the activity type
+     */
+    Intent(int id) {
+        this.id = id;
+    }
```

Comment:
Better have it as a parameter + field instead of using a switch statement.
```suggestion
    Intent(int id, boolean privileged) {
        this.id = id;
        this.privileged = privileged;
    }
```